code/datums/components/gas_leaker.dm

```diff
@@ -0,0 +1,59 @@
+#define PROCESS_PORTABLE "portable"
+#define PROCESS_COMPONENT "component"
+
+/datum/component/gas_leaker
+	var/process_type
+	var/integrity_leak_percent
+	var/leak_rate
+
+/datum/component/gas_leaker/Initialize(integrity_leak_percent=0.9, leak_rate=1)
+	. = ..()
+	if(istype(parent, /obj/machinery/portable_atmospherics))
+		process_type = PROCESS_PORTABLE
+	else if(istype(parent, /obj/machinery/atmospherics/components))
+		process_type = PROCESS_COMPONENT
+	else
+		return COMPONENT_INCOMPATIBLE
+
+	RegisterSignal(parent, COMSIG_OBJ_TAKE_DAMAGE, .proc/start_processing)
+
+	src.integrity_leak_percent = integrity_leak_percent
+	src.leak_rate = leak_rate
+
+/datum/component/gas_leaker/process()
+	. = ..()
+	switch(process_type)
+		if(PROCESS_PORTABLE)
+			. = process_portable()
+		if(PROCESS_COMPONENT)
+			. = process_component()
+		else
+			. = TRUE
+
+/datum/component/gas_leaker/proc/start_processing()
+	SIGNAL_HANDLER
+	// We're totally a pipe network, dont tell anyone
+	// This component needs to tick at the same rate as the atmos system
+	SSair.networks += src
```

Comment:
@LemonInTheDark think it's worth making another queue type for datums that wish to process at the same time as SSair?

Reply:
ahhhhhhhhhhhhhhhhhhhhh
If you need it make it yeah, if only so the performance of pipelines doesn't get muddled up with other stuff.
I think you might be able to pull this all off by tweaking the atmos sensitive element, but we can deal with that sort of thing later